File: minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int leng=(nums.length)/2;
       
        int i=0;
        int j=nums.length-1;
         int max=0;
        while(i<leng){
            int x= nums[i]+nums[j];
            max = Math.max(max,x);
            
            i++;
            j--;
        }
       
       
        return max;
        
    }
}

File: minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        max_sum=0
        l=0
        r=len(nums)-1
        nums.sort()
        while l<r:
            max_sum=max(max_sum,nums[l]+nums[r])
            l+=1
            r-=1
        return max_sum
            
